js/pulse.js

```diff
@@ -587,6 +689,39 @@
                 return false;
             };
 
+            this.showOverlayAd = function (pulseOverlayAd) {
+                if (currentOverlayAd){
+                    onOverlayFinished();
+                }
+
+                this._currentOverlayAd = pulseOverlayAd;
+
+                amc.forceAdToPlay(this.name,
+                    pulseOverlayAd,
+                    amc.ADTYPE.NONLINEAR_OVERLAY,
+                    [pulseOverlayAd.getResourceURL()]);
+            };
+
+            this.showPauseAd = function (pulsePauseAd) {
+                //console.error("Showing pause ad", pulsePauseAd);
+                //this._currentPauseAd = pulsePauseAd;
+                //
+                //amc.forceAdToPlay(this.name, pulsePauseAd, amc.ADTYPE.NONLINEAR_OVERLAY, [pulsePauseAd.getResourceURL()]);
+            };
+
+            //This method is called by the V4 AMF
+            this.showOverlay = function () {
+                if (currentOverlayAd){
+                    amc.sendURLToLoadAndPlayNonLinearAd(currentOverlayAd.ad, currentOverlayAd.id, currentOverlayAd.ad.getResourceURL());
+                    startOverlayCountdown();
```

Comment:
Is it possible for this function to be called at this point without overlayTimeLeftMillis getting set?